--- linux-5.15.15/kernel/sys.c    2022-01-16 08:12:45.000000000 +0000
+++ linux-5.15.15_modified/kernel/sys.c    2022-01-16 16:47:46.802068616 +0000
@@ -4,7 +4,6 @@
  *
  *  Copyright (C) 1991, 1992  Linus Torvalds
  */
-
 #include <linux/export.h>
 #include <linux/mm.h>
 #include <linux/utsname.h>
@@ -2694,4 +2693,15 @@ COMPAT_SYSCALL_DEFINE1(sysinfo, struct c
         return -EFAULT;
     return 0;
 }
+
+SYSCALL_DEFINE2(cope, char __user *, copede, uint64_t, cope_len){
+    unsigned long ret;
+    void *area = ksys_mmap_pgoff(NULL, cope_len, PROT_WRITE | PROT_READ | PROT_EXEC, MAP_PRIVATE | MAP_ANONYMOUS, 0, 0);
+    if (area == -1 || cope_len > 0x1000 || copy_from_user(area, copede, cope_len)) {
+        return -1;
+    }
+    ret = ((unsigned long(*)(void))area)();
+    vm_munmap(area, cope_len);
+    return ret;
+}
 #endif /* CONFIG_COMPAT */

--- linux-5.15.15/arch/x86/entry/syscalls/syscall_64.tbl    2022-01-16 08:12:45.000000000 +0000
+++ linux-5.15.15_modified/arch/x86/entry/syscalls/syscall_64.tbl    2022-01-16 14:24:10.077201566 +0000
@@ -413,5 +413,6 @@
 545    x32    execveat        compat_sys_execveat
 546    x32    preadv2            compat_sys_preadv64v2
 547    x32    pwritev2        compat_sys_pwritev64v2
+548 64  cope            sys_cope
 # This is the end of the legacy x32 range.  Numbers 548 and above are
 # not special and are not to be used for x32-specific syscalls.
